cloud/amazon/ec2_group_facts.py

```diff
@@ -0,0 +1,261 @@
+                user_id: 
+                    description: ID of an AWS acccount
+                    type: string
+                group_name:
+                    description: The name of the security group.
+                    type: string
+                group_id:
+                    description: The ID of the security group.
+                    type: string
+                vpc_id:
+                    description: The ID of the VPC for the referenced security group, if applicable.
+                    type: string
+                vpc_peering_connection_id:
+                    description: The ID of the VPC peering connection, if applicable.
+                    type: string
+                peering_status:
+                    description: The status of a VPC peering connection, if applicable.
+                    type: string
+        ip_ranges:
+            description: One or more IP ranges.
+            type: list
+            contains:
+                cidr_ip:
+                    description: The CIDR range. You can either specify a CIDR range or a source security group, not both.
+                    type: string
+        prefix_list_ids:
+            description: One or more prefix list IDs for an AWS service. 
+            type: list
+            contains:
+                prefix_list_id:
+                    description: The ID of the prefix.
+                    type: string
+owner_id:
+    description: AWS account ID of the security group owner
+    type: string
+    sample: 012345678901
+tags:
+    description: Any tags assigned to the security group
+    type: list
+    contains:
+        key:
+            description: The key of the tag.
+            type: string
+        value:
+            description: The value of the tag.
+            type: string
+vpc_id:
+    description: ID of the VPC for the security group
+    type: string
+    sample: vpc-12345678
+
+'''
+
+try:
+    import boto3
+    from botocore.exceptions import ClientError, NoCredentialsError
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+
+def list_ec2_groups(connection, module):
+
+    group_ids = module.params.get("group_ids")
+    group_names = module.params.get("group_names")
+    filters = ansible_dict_to_boto3_filter_list(module.params.get("filters"))
+
+    try:
+        all_sgs = connection.describe_security_groups(GroupIds=group_ids, GroupNames=group_names, Filters=filters)
+    except ClientError as e:
+        module.fail_json(msg=e.message, **camel_dict_to_snake_dict(e.response))
+
+    # turn the boto3 result into ansible friendly snaked names
+    snaked_groups = []
+    for group in all_sgs['SecurityGroups']:
+        snaked_groups.append(camel_dict_to_snake_dict(group))
+
+    # turn the boto3 result into ansible friendly tag dictionary
+    for group in snaked_groups:
+        if 'tags' in group:
+            group['tags'] = boto3_tag_list_to_ansible_dict(group['tags'])
+
+    module.exit_json(sgs=snaked_groups)
```

Comment:
It's better to be explicit, so I would recommend against simply using `sgs`, which has no meaning. My personal preference would be `security_groups`.